cmd/deletechainsfinalizer/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"flag"
+	tektonclientset "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"path/filepath"
+	"strings"
+)
+
+// There is currently a bug in tekton chains that can prevent pipeline runs from being deleted
+// Run this from the IDE to fix it
+// Do not use this in production
+func main() {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kube", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kube", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	plnClient, err := tektonclientset.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+	namespaceList, err := clientset.CoreV1().Namespaces().List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return
+	}
+	for _, namespace := range namespaceList.Items {
+		println("Namespace: " + namespace.Name)
+		if strings.Contains(namespace.Name, "jvm") || strings.Contains(namespace.Name, "kcp") {
```

Comment:
so do you run this against the cluster kubeconfig, kcp kubeconfig, either, or both?

Reply:
Just against compute.

The first issue I ran into was fixed by updating the image, this needs to be fixed upstream.